src/sync/remote_mongo_client.cpp

```diff
@@ -17,3 +17,20 @@
 ////////////////////////////////////////////////////////////////////////////
 
 #include "sync/remote_mongo_client.hpp"
+#include "sync/remote_mongo_database.hpp"
```

Comment:
Here and for the `remote_mongo_client.hpp`. It would be nice if we could expose the names as public arguments.